framework/src/plugins/base_plugin.ts

```diff
@@ -252,41 +251,31 @@ export abstract class BasePlugin<
 					this.schemas.transactionsAssets,
 				),
 		};
-	}
 
-	// eslint-disable-next-line @typescript-eslint/require-await
-	public async init(channel: BaseChannel): Promise<void> {
 		const dirs = systemDirs(this.options.appConfig.label, this.options.appConfig.rootPath);
 		this._logger = createLogger({
 			consoleLogLevel: this.options.appConfig.logger.consoleLogLevel,
 			fileLogLevel: this.options.appConfig.logger.fileLogLevel,
 			logFilePath: join(
 				dirs.logs,
-				`plugin-${((this.constructor as unknown) as typeof BasePlugin).alias}.log`,
+				`plugin-${((this.constructor as unknown) as typeof BasePlugin).name}.log`,
 			),
-			module: `plugin:${((this.constructor as unknown) as typeof BasePlugin).alias}`,
+			module: `plugin:${((this.constructor as unknown) as typeof BasePlugin).name}`,
 		});
 
-		channel.once(APP_EVENT_READY, async () => {
-			this.schemas = await channel.invoke('app:getSchema');
+		context.channel.once(APP_EVENT_READY, async () => {
+			this.schemas = await context.channel.invoke('app:getSchema');
 		});
 	}
 
-	// eslint-disable-next-line @typescript-eslint/no-empty-function
-	public static get alias(): string {
-		throw new ImplementationMissingError();
-	}
-	// eslint-disable-next-line @typescript-eslint/no-empty-function
-	public static get info(): PluginInfo {
-		throw new ImplementationMissingError();
-	}
-
 	// TODO: To make non-breaking change we have to keep "object" here
-	public get defaults(): SchemaWithDefault | object | undefined {
+	public get configSchema(): SchemaWithDefault | object | undefined {
 		return undefined;
 	}
+	public abstract get nodeModulePath(): string;
 	public abstract get events(): EventsDefinition;
 	public abstract get actions(): ActionsDefinition;
```

Comment:
I will suggest to remove these from `abstract` and not make these required for plugin developer to define. One can just develop a plugin without defining events and actions.

Reply:
I will create issue to make properties to be consistent